django/contrib/admin/options.py

```diff
@@ -561,14 +577,33 @@ def wrapper(*args, **kwargs):
         urlpatterns = [
             url(r'^$', wrap(self.changelist_view), name='%s_%s_changelist' % info),
             url(r'^add/$', wrap(self.add_view), name='%s_%s_add' % info),
+            url(r'^autocomplete/(?P<field_name>.+)/$',
+                wrap(AutocompleteJsonView.as_view(
+                    admin_site=self.admin_site,
+                    source_model=self.model,
+                    source_model_admin=self,
+                )), name='%s_%s_autocomplete' % info),
             url(r'^(.+)/history/$', wrap(self.history_view), name='%s_%s_history' % info),
             url(r'^(.+)/delete/$', wrap(self.delete_view), name='%s_%s_delete' % info),
             url(r'^(.+)/change/$', wrap(self.change_view), name='%s_%s_change' % info),
+        ]
+        for inline in self.inlines:
+            inline_info = inline.model._meta.app_label, inline.model._meta.model_name
+            if inline.autocomplete_fields:
+                urlpatterns.append(
+                    url(r'^autocomplete_inline/(?P<field_name>.+)/$',
```

Comment:
this would register the same url multiple times in case of more than one inline, no?